crates/composer/src/jukebox.rs

```diff
@@ -49,12 +49,17 @@ impl Jukebox {
         Ok(Self { samples })
     }
 
-    pub(crate) fn play(&self, audio_output: &AudioOutput, sample: Sample) {
+    pub(crate) fn play(
+        &self,
+        audio_output: &AudioOutput,
+        sample: Sample,
+        timestamp: Option<Duration>,
+    ) {
         let buffer = self
             .samples
             .get(&sample)
             .expect("programmer error, all possible samples should be loaded");
 
-        audio_output.play(buffer.clone());
+        audio_output.play(buffer.clone(), timestamp);
```

Comment:
I don't like how we need to pass the timestamp through the jukebox. Maybe we could gut the jukebox out even more and instead of `play` that takes the `audio_output` we could have
```rust
get_sound(sample: Sample) -> Buffer
```

and play pass the returned buffer to the `audio_output` at the cal site?

Reply:
I put that to side into an issue: https://github.com/tonarino/acoustic_profiler/issues/43